standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/AcidEventListener.java

```diff
@@ -72,6 +77,22 @@ public void onDropPartition(DropPartitionEvent partitionEvent)  throws MetaExcep
       txnHandler = getTxnHandler();
       txnHandler.cleanupRecords(HiveObjectType.PARTITION, null, partitionEvent.getTable(),
           partitionEvent.getPartitionIterator());
+
+      if (MetastoreConf.getBoolVar(conf, ConfVars.LOCKLESS_READS_ENABLED)) {
+        CompactionRequest rqst = new CompactionRequest(partitionEvent.getTable().getDbName(), partitionEvent.getTable().getTableName(),
```

Comment:
Is this going to work? If the partition record is dropped, I think the compaction will automatically fail, there should be a test for this
I think you should just create a compaction record that is in ready_for_cleaning state. And since there is a base file, I think I would prefer a major compaction, but it does not really matter.